proposals/collection-literals.md

```diff
@@ -112,23 +112,48 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
 
     * The behavior of collection literals with collections that are not well-behaved is undefined.
 
-## Constructible collection types (target types)
-[constructible-collection-types]: #constructible-collection-types
+## Conversions
+A collection literal expression is classified as a value.
 
-The following are _constructible collection types_ that can be used to construct collection literals.
-These are the valid _target types_ for collection literals.
+A _collection literal conversion_ permits a collection literal expression to be implicitly converted to a type.
 
-The list is in priority order. If a collection type belongs in multiple categories, the first is used.
+The following collection literal conversions exist from a collection literal:
 
-Actual translation of the literal to the corresponding type is defined [below](#collection-literal-translation).
+* To a single dimensional *array type* `T[]`, or a *span type* `System.Span<T>` or `System.ReadOnlySpan<T>`, where for each element `Ei`:
+  * If `Ei` is an *expression_element* there is an implicit conversion from `Ei` to `T`.
+  * If `Ei` is a *dictionary_element* `Ki:Vi`, then `T` is a _type_ `System.Collections.Generic.KeyValuePair<K, V>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *spread_element* with enumerated element type `Si` there is an implicit conversion from `Si` to `T`.
 
-* Single dimensional arrays (e.g. `T[]`)
-* [*Span types*](#span-types)
-* Types with a suitable [`Construct` method](#construct-methods)
-* Types that implement `System.Collections.IDictionary` - *dictionary [collection initializers](#collection-initializers)*
-* Interface types _`I<TKey, TValue>`_ implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
-* Types that implement `System.Collections.IEnumerable` - [*collection initializers*](#collection-initializers)
-* Interface types _`I<T>`_ implemented by `System.Collections.Generic.List<T>` (e.g. `IEnumerable<T>`, `IList<T>`, `IReadOnlyList<T>`)
+* To a *type* with an associated *[builder](#construct-methods)* where there is an implicit collection literal conversion from the collection literal to the *builder argument type*.
+
+* To a *type* that implements `System.Collections.IDictionary` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter `int capacity`.
+  * For each element `Ei` there is an applicable instance indexer for the *key_value_pair* from:
+    * If `Ei` is an *expression_element* with type `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from `Ei`.
```

Comment:
you mention 'dynamic' here but not for the array/span cases.  intentional/necessary?

Reply:
The reason for mentioning `dynamic` here is we'll use `Key` and `Value` properties, and we want to limit the set of types that are considered *key value pairs*. It feels like `dynamic` might be too much because that would allow types other than `KeyValuePair<,>`.